### ISSUE TITLE
excluded nodes still considered in scheduler feasibility check

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/flux-core#3757]
From our ☕ call today, if you exclude a node at runtime (not at configure time), then the scheduler will only mark the node as `DOWN` and not remove/exclude it entirely. Meaning that feasibility checks that should fail (e.g., requesting every node in the system, including the excluded one) will not fail.

IIUC, solving this requires a modification to the `resource` <-> `sched` communication such that `DOWN` is differentiated from `EXCLUDED`.  For sched-simple and Fluxion, we would then need to decide how the `EXCLUDED` message is handled.  Are the resources removed from the schedulers internal data structures, or just excluded from feasibility checks?

**Comment by garlick** [comment]
See also flux-framework/rfc#253 (my bad for not finishing that!)

Also as a fallback:   we _should_ be able to `flux module reload sched-simple` (or fluxion equivalent) without affecting running jobs.  This would cause the resource protocol to start from scratch, with the new excluded set omitted from the resource definition.

**Comment by dongahn** [comment]
Also https://github.com/flux-framework/flux-sched/issues/842 -- related. Try a different policy for testing.

**Comment by grondo** [comment]
> Also as a fallback: we should be able to flux module reload sched-simple (or fluxion equivalent) without affecting running jobs. This would cause the resource protocol to start from scratch, with the new excluded set omitted from the resource definition.

May have to be careful to do this _after_ the newly-excluded node has been drained and no more jobs have that resource allocated. Otherwise, an alloc event for a running job that contains the excluded node could confuse the scheduler.

**Comment by garlick** [comment]
Good point - did you mean s/alloc/free/?

**Comment by grondo** [comment]
Yeah, sorry! I was actually thinking of the _R_ fetched as a result of the `hello` protocol for each outstanding job ([RFC 27](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_27.html#hello)). The `free` request would be equally problematic I assume.
